### ISSUE TITLE
Idle Sleep Interrupt Implementation

### ISSUE THREAD

**Issue by garudaonekh** [Spirik/GEM#107]
Hi,
I need to the RGB TFT to go to sleep after around 20 seconds of idle time and wakeup upon any button clicked.
I notice the Keydetect is not based on interrupt. So I guess there is no existing such features in the GEM, thus I have to detect this manually and add interrupt to all the key. and redraw the menu upon wakeup.

Thanks;

**Comment by Spirik** [comment]
Yes, you are correct. GEM doesn't rely on any kind of interrupts internally. The way user interacts with it is up to the author of the sketch. You can implement your own key detection routine as you seem fit (KeyDetector is a separate library and is not integral part of the GEM itself). As long as you call [registerKeyPress()](https://github.com/Spirik/GEM/wiki/How-to#control-menu-from-your-sketch) and `drawMenu()` methods of GEM in a proper time, menu should be operational regardless of exact input method.
